File: Python/model_boosting_xgboost.py
# -*- coding: utf-8 -*-
"""
ML Lab for the udemy course "Machine Learning A-Z (Codes and Datasets)"
Topic: Model Selection and Boosting- XGBoost
@author: Anand
"""
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix

from sklearn.model_selection import train_test_split

from xgboost import XGBClassifier

from sklearn.model_selection import cross_val_score

dataset = pd.read_csv("../Data/Xgboost_Data.csv")


X = dataset.iloc[:,:-1] #Selecting all but the last columns of the dataset
y = dataset.iloc[:,-1] #Selecting the last column of the dataset

X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.2, random_state=0)


#Training the XGBoost model
classifier = XGBClassifier()
classifier.fit(X_train,y_train)


y_pred = classifier.predict(X_test)

#Computing the accuracy and confusion matrix for the model
accuracy = accuracy_score(y_test, y_pred)
cm = confusion_matrix(y_test, y_pred)


#Implement K-Fold Cross Validation#
accuracies = cross_val_score(estimator=classifier, X=X_train, y=y_train)
print("Accuracy :{:.2f} % ".format(accuracies.mean() * 100))
print("Standard Deviation :{:.2f} % ".format(accuracies.std() * 100))

File: R/model_boosting_xgboost.R
#' 
#' ML Lab for the udemy course "Machine Learning A-Z (Codes and Datasets)"
#' Topic: XGBoost

#Import the dataset#
dataset <- read.csv("../Data/Churn_Modelling.csv")
dataset = dataset[,4:14]

#Encoding the categorical values#
levels <- unique(dataset$Geography)
labels <- 1 : (length(levels))
dataset$Geography = as.numeric(factor(dataset$Geography, levels = levels, labels = labels))
levels <- unique(dataset$Gender)
labels <- 0 : (length(levels) - 1)
dataset$Gender = as.numeric(factor(dataset$Gender, levels = levels, labels = labels))


#Splitting the data into train and test#
library(caTools)
set.seed(6)
split = sample.split(dataset$Exited, SplitRatio = .8)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

#Building the XGBoost#
library(xgboost)
classifier <- xgboost(data = as.matrix(training_set[-11]), label = training_set$Exited, nrounds = 10)


#Predicting the probabilities
y_pred = as.data.frame(predict(classifier, newdata = as.matrix(test_set[-11])))
y_pred$predict <- ifelse(y_pred$predict > 0.5, 1, 0)
#Computiing the model characteristics#
cm = table(test_set[,11], y_pred$predict)
